test_runner/performance/test_physical_replication.py

```diff
@@ -0,0 +1,288 @@
+                    zenbenchmark.record("replica_lag", lag, "s", MetricReport.LOWER_IS_BETTER)
+            finally:
+                replica_workload.terminate()
+        finally:
+            master_workload.terminate()
+    except Exception:
+        should_delete = False
+    finally:
+        if should_delete:
+            neon_delete_project(neon_api_key, neon_api_base_url, project_id)
+
+
+def report_pgbench_aggregate_intervals(
+    output_dir: Path,
+    prefix: str,
+    zenbenchmark: NeonBenchmarker,
+):
+    for filename in os.listdir(output_dir):
+        if filename.startswith(prefix):
+            # The file will be in the form <prefix>_<node>.<pid>
+            # So we first lop off the .<pid>, and then lop off the prefix and the _
+            node = filename.split(".")[0][len(prefix) + 1 :]
+            with open(output_dir / filename) as f:
+                reader = csv.reader(f, delimiter=" ")
+                for line in reader:
+                    num_transactions = int(line[1])
+                    if num_transactions == 0:
+                        continue
+                    sum_latency = int(line[2])
+                    sum_lag = int(line[3])
+                    zenbenchmark.record(
+                        f"{node}_num_txns", num_transactions, "txns", MetricReport.HIGHER_IS_BETTER
+                    )
+                    zenbenchmark.record(
+                        f"{node}_avg_latency",
+                        sum_latency / num_transactions,
+                        "s",
+                        MetricReport.LOWER_IS_BETTER,
+                    )
+                    zenbenchmark.record(
+                        f"{node}_avg_lag",
+                        sum_lag / num_transactions,
+                        "s",
+                        MetricReport.LOWER_IS_BETTER,
+                    )
+
+
+@pytest.mark.remote_cluster
+@pytest.mark.timeout(2 * 60 * 60)
+def test_replication_start_stop(
+    pg_bin: PgBin,
+    test_output_dir: Path,
+    neon_api_key: str,
+    neon_api_base_url: str,
+    pg_version: PgVersion,
+    zenbenchmark: NeonBenchmarker,
+):
+    """
+    Cycles through different configurations of read replicas being enabled disabled. The whole time,
+    there's a pgbench read/write workload going on the master. For each replica, we either turn it
+    on or off, and see how long it takes to catch up after some set amount of time of replicating
+    the pgbench.
+    """
+
+    prefix = "pgbench_agg"
+    num_replicas = 2
+    configuration_test_time_sec = 10 * 60
+    should_delete = True
+
+    project = neon_create_project(neon_api_key, neon_api_base_url, pg_version)
+    project_id = project["project"]["id"]
+    neon_wait_for_operation_to_finish(neon_api_key, neon_api_base_url, project_id)
+    try:
+        branch_id = project["branch"]["id"]
+        master_connstr = project["connection_uris"][0]["connection_uri"]
```

Comment:
This connection string contains a password to the DB, and will be exposed to the internet in CI logs.
Usually, we set password through `PGPASSWORD` env var, see examples in:
https://github.com/neondatabase/neon/blob/3e63d0f9e0532d0e49668bf41b2534ffcee0ca2a/test_runner/performance/test_perf_pgbench.py#L58
and in
https://github.com/neondatabase/neon/blob/3e63d0f9e0532d0e49668bf41b2534ffcee0ca2a/test_runner/performance/test_perf_olap.py#L147

We also might want to mask password from logs:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-masking-a-string
(using only masking is not enough, because Allure reports won't have such values masked in logs)

Reply:
Switched to environments